ingestion/src/metadata/ingestion/stage/file.py

```diff
@@ -56,8 +55,11 @@ def create(
         metadata_config = MetadataServerConfig.parse_obj(metadata_config_dict)
         return cls(ctx, config, metadata_config)
 
-    def stage_record(self, record: Table) -> None:
-        json_record = json.loads(record.json())
+    def stage_record(self, record) -> None:
+        if hasattr(record, "to_json"):
```

Comment:
we are using pydantic pretty much everywhere. Do you see any places where we may need to_json?

Reply:
sample_data.py uses ```ingestion/src/metadata/ingestion/models/user.py``` and not ```ingestion/src/metadata/generated/schema/entity/teams/user.py```. The former has "to_json" and the latter has "json". I guess we should eventually replace the former with the latter. I'm not familiar with ```pydantic``` yet. In this case, ```record``` can be almost any generated class (Table, Chart, and so on) and not just Table.

Reply:
@amiorin makes sense. At least for user.py in model we should remove that and use the generated classes for these. cc @ayush-shah 
Agree with record being any generated class. I am +1 with going what you have. 
But we should file a follow up ticket and make the record can only be any generated or self-created pydantic , that way we can always depend on record.json(). Also to_json() something we added if we allow users to pass any class we may not have to_json() or json() itself.

Reply:
We might be able to achieve this with a `TypeVar` bound to `BaseModel`

Reply:
btw , there is patch that's moving user and team to Pydantic. https://github.com/open-metadata/OpenMetadata/pull/1473 . So we can remove this and make sure the record is of Pydantic. 
@pmbrull any suggestion on how to enforce this would be great . We can handle this another PR

Reply:
Was gonna share Omri's PR as well, thanks! Opened https://github.com/open-metadata/OpenMetadata/issues/1484 to further generalise the implementation.

Reply:
@amiorin lets remove this change. Since we are handling it with pydantic

Reply:
I have removed this change.